packages/react-dom/src/server/ReactPartialRenderer.js

```diff
@@ -542,7 +542,11 @@ function resolve(
         if (typeof Component.getDerivedStateFromProps !== 'function') {
           inst.componentWillMount();
         }
-      } else if (typeof Component.getDerivedStateFromProps !== 'function') {
+      }
+      if (
+        inst.UNSAFE_componentWillMount &&
```

Comment:
Maybe change this to test for `function` type explicitly for consistency? I know it didn't before but still.